### DISCUSSION TITLE
does unit-work help ?

### DISCUSSION THREAD

**Discussion by superiums** [mikro-orm/mikro-orm#4537] [Q&A] [answered]
I'm sorry that i'm puzzeld for the unit-work of mikroorm in web request,
eg.

```
const user = await em.findOneOrFail(User, 1);
// ---> we return user to web client this time.

//----> then we let client to modify it, so these changes was done in client, and later, we receive the change by another client request.
user.email = 'foo@bar.com';
const car = new Car();
user.cars.add(car);

// ---> so in this situation, does unit-work work ?  is it neccessary to keep an cache of queried object ?
await em.flush();

```

i'm a little puzzeld.

**Comment by B4nan** [comment]
Please use discussions for stuff like this, how is this a feature request?

**Reply by superiums** [reply]
ok, i'm sorry that not find the Discussion was avaluable.

**Comment by B4nan** [comment]
The example you provided sounds like 3 different requests, those should have their own contexts - 3 requests, 3 different contexts/identity maps. 

The identity map is there to help you mostly in complex apps, where you have logic spread across multiple levels, services, and whatnot. It helps you to not override changes from one place in another - but still in one context, in one request handler, not among more of them - those should be always isolated.

To get around issues with parallel requests trying to modify an entity you can use optimistic locking.

https://mikro-orm.io/docs/next/transactions#optimistic-locking

**Comment by superiums** [comment]
am I got it ?
1. unit-work only works in ONE request context, not multi request. so in fact we dont' need it in normal CRUD service. like this: 
`client -->controller-->service -->controller-->client.`
2. unit-work was designed to work with complex serices. like this :  
 `client--->controller-->serviceA-->serviceB--->serviceC....-->controller-->client`.

so do we need to disable cache on simple CRUD ?

**Reply by B4nan** [reply, marked as answer]
The unit of work pattern is not about caching, it's about grouping changes together into a single transaction (and rolling them back as a whole if something breaks). Flushing works solely by working with the entity state, if you disable the identity map, you won't be able to modify your entities via the flushing mechanism.

**Reply by superiums** [reply]
got it, thanks very much !
